vm/src/obj/objbytearray.rs

```diff
@@ -92,6 +132,99 @@ fn bytearray_eq(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
     Ok(vm.ctx.new_bool(result))
 }
 
+fn bytearray_isalnum(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(vm, args, required = [(zelf, Some(vm.ctx.bytearray_type()))]);
+    let bytes = get_value(zelf);
+    Ok(vm.new_bool(!bytes.is_empty() && bytes.iter().all(|x| char::from(*x).is_alphanumeric())))
```

Comment:
I have the feeling that we need functions like `chars_isalnum` which take an `Iter<char>` argument. Then we can use this function for `str`, `bytes` and `bytearray`.